docs/pages/kubernetes-access/controls.mdx

```diff
@@ -242,29 +126,81 @@ Label each cluster with key-value pairs describing the cluster:
+uses  [impersonation
+headers](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#user-impersonation)
+to send requests to the API server with one Kubernetes user and zero or more
+Kubernetes groups.
+
+![Impersonation](../../img/k8s/auth.svg)
+
+The `kubernetes_users` and `kubernetes_groups` fields indicate which users and
+groups to allow a user to assume when they send requests to a Kubernetes API
+server:
 
 ```yaml
 kind: role
-version: v5
 metadata:
-  name: group-member
+  name: kube-access
+version: v6
 spec:
   allow:
-    # Allow access to Kubernetes clusters in the production environment
-    kubernetes_labels:
-      'env': 'prod'
-    # Allows access to group "admins" in Kubernetes clusters
-    kubernetes_groups: ["admins"]
+    kubernetes_groups:
+    - developers
+    - viewers
+    kubernetes_users:
+    - myuser
+    # ...
+  deny: {}
 ```
 
-## Impersonation
+The value of `kubernetes_groups` and `kubernetes_users` is a list of names of
+groups and users to enable impersonation for.
+
+#### How Teleport users impersonate Kubernetes users and groups
+
+There are two ways for an end user to specify which user and groups to
+impersonate:
+
+##### Manually
+
+When a user runs `tsh kube login` to authenticate to a Kubernetes cluster, they
+can use the `--as` and `--as-groups` flags to manually specify the user and
+groups to authenticate as. The Teleport Kubernetes Service determines whether
+the user and groups belong to a user's `kubernetes_users` and
+`kubernetes_groups` configuration and, if not, denies the user access.
+
+##### Automatically
 
-Teleport uses [Kubernetes impersonation](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#user-impersonation)
-to map OIDC and SAML users to Kubernetes users and groups.
+If the user has not explicitly determined a Kubernetes user and Kubernetes
+groups when authenticating to a cluster, the Teleport Kubernetes Service
+determines this from the `kubernetes_users` and `kubernetes_groups` fields in a
+user's roles.
 
-Here is an example of Kubernetes `ClusterRole` and `ClusterRoleBinding`
-allowing Teleport's `ServiceAccount` to impersonate users and groups:
+If a user has exactly one value in `kubernetes_users`, the Teleport Kubernetes
+Service impersonates that user. Otherwise, the Kubernetes Service denies the
+user access. In this case, the Teleport user must manually specify a Kubernetes
+user to impersonate.
+
+If the user has not specified a Kubernetes group to impersonate, the Kubernetes
+Service uses all values within `kubernetes_groups`.
+
+With the `kube-access` role above, after you authenticate to Teleport, the
+Kubernetes Service uses impersonation headers to forward requests to the API
+server with the `developers` group and the `myuser` Kubernetes user.
+
+#### Enabling impersonation
+
+To enable the Kubernetes Service to forward user requests with impersonation
```

Comment:
Mention the script that creates the SA automatically with the roles, so users don't try to recreate this

Reply:
I moved the `Notice` that was below the snippet above it so users see it before the snippet.